GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/IncludedFolderTests.cs

```diff
@@ -0,0 +1,231 @@
+            this.ValidateIncludedFolders(this.mainIncludedFolder);
+
+            string newFolderPath = Path.Combine(this.Enlistment.RepoRoot, "GVFS", "GVFS.Common");
+            newFolderPath.ShouldNotExistOnDisk(this.fileSystem);
+            Directory.CreateDirectory(newFolderPath);
+            newFolderPath.ShouldBeADirectory(this.fileSystem);
+            string[] fileSystemEntries = Directory.GetFileSystemEntries(newFolderPath);
+            fileSystemEntries.Length.ShouldEqual(32);
+
+            string projectedFolder = Path.Combine(newFolderPath, "Git");
+            projectedFolder.ShouldBeADirectory(this.fileSystem);
+            fileSystemEntries = Directory.GetFileSystemEntries(projectedFolder);
+            fileSystemEntries.Length.ShouldEqual(13);
+
+            string projectedFile = Path.Combine(newFolderPath, "ReturnCode.cs");
+            projectedFile.ShouldBeAFile(this.fileSystem);
+        }
+
+        [TestCase, Order(6)]
+        public void ReadFileThenChangingIncludeFoldersShouldRemoveFileAndFolder()
+        {
+            string fileToRead = Path.Combine(this.Enlistment.RepoRoot, "Scripts", "RunFunctionalTests.bat");
+            this.fileSystem.ReadAllText(fileToRead);
+
+            this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder);
+            this.ValidateIncludedFolders(this.mainIncludedFolder);
+
+            string folderPath = Path.Combine(this.Enlistment.RepoRoot, "Scripts");
+            folderPath.ShouldNotExistOnDisk(this.fileSystem);
+            fileToRead.ShouldNotExistOnDisk(this.fileSystem);
+        }
+
+        [TestCase, Order(7)]
+        public void CreateNewFileWillPreventRemoveIncludedFolder()
+        {
+            this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder, "Scripts");
+            this.ValidateIncludedFolders(this.mainIncludedFolder, "Scripts");
+
+            string fileToCreate = Path.Combine(this.Enlistment.RepoRoot, "Scripts", "newfile.txt");
+            this.fileSystem.WriteAllText(fileToCreate, "New Contents");
+
+            this.gvfsProcess.RemoveIncludedFolders("Scripts");
+            this.ValidateIncludedFolders(this.mainIncludedFolder, "Scripts");
+
+            string folderPath = Path.Combine(this.Enlistment.RepoRoot, "Scripts");
+            folderPath.ShouldBeADirectory(this.fileSystem);
+            string[] fileSystemEntries = Directory.GetFileSystemEntries(folderPath);
+            fileSystemEntries.Length.ShouldEqual(6);
+            fileToCreate.ShouldBeAFile(this.fileSystem);
+
+            this.fileSystem.DeleteFile(fileToCreate);
+        }
+
+        [TestCase, Order(8)]
+        public void ModifiedFileShouldNotAllowIncludedFolderChange()
+        {
+            string modifiedPath = Path.Combine(this.Enlistment.RepoRoot, "Scripts", "RunFunctionalTests.bat");
+            this.fileSystem.WriteAllText(modifiedPath, "New Contents");
+
+            this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder);

```

Comment:
Would we expect an error here (if gvfsProcess was not allowed to actually add an included folder)? Or maybe the actual return text of running this command does indicate there was an error.